app/src/main/java/org/simple/clinic/user/refreshuser/RefreshCurrentUser.kt

```diff
@@ -0,0 +1,68 @@
+package org.simple.clinic.user.refreshuser
+
+import io.reactivex.Completable
+import io.reactivex.Single
+import org.simple.clinic.user.LoggedInUserPayload
+import org.simple.clinic.user.User
+import org.simple.clinic.user.User.LoggedInStatus
+import org.simple.clinic.user.User.LoggedInStatus.LOGGED_IN
+import org.simple.clinic.user.User.LoggedInStatus.RESET_PIN_REQUESTED
+import org.simple.clinic.user.User.RoomDao
+import org.simple.clinic.user.UserStatus
+import org.simple.clinic.user.finduser.FindUserResult.Found
+import org.simple.clinic.user.finduser.FindUserWithPhoneNumber
+import org.simple.clinic.util.mapType
+import timber.log.Timber
+
+class RefreshCurrentUser(
+    private val userDao: RoomDao,
+    private val findUserWithPhoneNumber: FindUserWithPhoneNumber
+) {
+
+  fun refresh(): Completable {
+    return Single.fromCallable { userDao.userImmediate() }
+        .doOnSuccess { Timber.i("Refreshing logged-in user") }
+        .flatMapCompletable { user ->
+          findUserWithPhoneNumber.find(user.phoneNumber)
+              .mapType<Found, LoggedInUserPayload> { it.user }
```

Comment:
Curious, are error cases handled by the caller?

Reply:
Currently, yes. In our current implementation, the only caller (`PatientsScreenController`) ignores any error that was generated since this is a fire-and-forget operation.

Reply:
Fire and forget isn't explicit?

Reply:
It's explicit at the usage site, i.e., in the `PatientsScreenController`.